### PR TITLE
Valet Reservation refresh button + redux refactoring 

### PR DESCRIPTION
**Pull request by kjmak14**
This PR has 3 main changes:

1. Refactored the CurrentSpots and ReservationList to redux; also implemented useCallback and useMemo
2. Combined all the valet reservation util files into one file and deleted the extra files
3. Implemented a refresh button on  the valet reservation page that fetches both the spot information and reservation list given a specific date, garage_id and time. 

This PR might seem large, but it is mainly relocating the fetchReservations and fetchSpots logic from their respective components to the ValetReservation container. There is still a lot of additional refactoring to be done and I left console.logs in some of the valet reservation components for testing purposes.

### CODE REVIEW

File: src/components/valet-reservation/CurrentSpots.jsx [hunk 1]
```diff
@@ -1,55 +1,25 @@
...
-        setOccupiedSpots(occupied);
-      } catch (error) {
-        console.log(error);
-      }
-    };
-
-    useEffect(() => {
-      fetchSpots();
-    }, []);
+    const { date, spots } = useSelector((state) => state.valet);
```
Comment (bzw24) [inline src/components/valet-reservation/CurrentSpots.jsx:6]: What is the purpose of this useEffect, is it just for troubleshooting purposes?

File: src/components/valet-reservation/ReservationList.jsx [hunk 2]
```diff
@@ -22,23 +16,10 @@ const ReservationList = () => {
...
-      dispatch(setReservations(data))
-    } catch (error) {
-      console.log(error);
-    }
-  }, [garage_id, date, dispatch]);
-
-  useEffect(() => {
-    fetchReservations();
-  }, [time, fetchReservations]);
-
```
Comment (bzw24) [inline src/components/valet-reservation/ReservationList.jsx:37]: Also again, what is this useEffect doing?

File: src/containers/ValetReservation.jsx [hunk 3]
```diff
@@ -62,6 +86,11 @@ const ValetReservation = () => {
       <button onClick={() => handleGetQrCode(event)}>Show QR Code </button>
       {showQrCode && qrCodeLink ? <img src={qrCodeLink} /> : null}
       <Link to="/scanner">Scanner</Link>
```
Comment (bzw24) [inline src/containers/ValetReservation.jsx:88]: can delete the xmlns link for future refactor. I guess so we are all aligned for icons.

File: src/utils/valetReservationUtils.js
Comment (bzw24) [inline src/utils/valetReservationUtils.js:1]: good putting all axios on the same file

File: src/components/valet-reservation/ReservationList.jsx [hunk 4]
```diff
@@ -1,16 +1,10 @@
...
-import getReservationList from "../../utils/getReservationList";
 
 const ReservationList = () => {
   const [showNextHour, setShowNextHour] = useState(false);
-  const { time, reservations } = useSelector((state) => state.valet);
-  const dispatch = useDispatch();
-
-  const garage_id = 1;
-  const date = "12-31-23";
+  const { time, reservations, date } = useSelector((state) => state.valet);
```
Comment (bzw24) [inline src/components/valet-reservation/ReservationList.jsx:8]: good use of memo for optimization.

**Review by bzw24 — APPROVED** [review]

File: src/containers/ValetReservation.jsx [hunk 5]
```diff
@@ -2,21 +2,40 @@
...
+      const spotData = await getSpotMetrics(garage_id, date, time);
+      const reservationData = await getReservationList(garage_id, date);
+      dispatch(setSpots(spotData));
+      dispatch(setReservations(reservationData))
+    } catch (error) {
+      console.log(error);
+    }
+  }, [garage_id, date, time, dispatch])
+
+  // const fetchReservations = useCallback(async () => {
```
Comment (jose-m-f) [inline src/containers/ValetReservation.jsx:30]: If this code isn't necessary I think you should just delete it. You can do this in a future PR if you're testing something first and aren't sure if you're getting rid of it.

**Review by kssampson — APPROVED** [review]
I know we chatted about including a commit to remove the attempted calendar picker in this, but not needed/included here so excluding sounds good. Thanks for the refactor.

File: src/components/valet-reservation/ReservationList.jsx [hunk 6]
```diff
@@ -22,23 +16,10 @@ const ReservationList = () => {
...
-
-  useEffect(() => {
-    fetchReservations();
-  }, [time, fetchReservations]);
-
   useEffect(() => {
-    const interval = setInterval(() => dispatch(setTime()), 1000 * 60 * 60); //check every hour
-    return () => clearInterval(interval);
-  }, [dispatch]);
+    console.log('checking date in list: ', date);
```
Comment (jose-m-f) [inline src/components/valet-reservation/ReservationList.jsx:20]: These logs should be removed if you're no longer testing things.

File: src/utils/valetReservationUtils.js
Comment (jose-m-f) [inline src/utils/valetReservationUtils.js:1]: I agree with Bruce, this makes it much easier to understand your data flow

**Review by jose-m-f — APPROVED** [review]
Hey Kyle, I left a couple of comments that you can address in a future PR

File: src/components/valet-reservation/CurrentSpots.jsx (hunk 1 above)
Comment (kjmak14) [reply src/components/valet-reservation/CurrentSpots.jsx:6]: Yes, just for troubleshooting. Will remove later

File: src/components/valet-reservation/ReservationList.jsx (hunk 6 above)
Comment (kjmak14) [reply src/components/valet-reservation/ReservationList.jsx:20]: Still testing some things, but yes I will remove them

File: src/containers/ValetReservation.jsx (hunk 5 above)
Comment (kjmak14) [reply src/containers/ValetReservation.jsx:30]: Good point!

File: src/containers/ValetReservation.jsx (hunk 3 above)
Comment (kjmak14) [reply src/containers/ValetReservation.jsx:88]: Yup I'm leaving it in until we come together for the CSS refactor

**Pull request merged**